### PR TITLE
Add HW20

### PR DESCRIPTION
**Pull request by Gritskov-Vitaliy**

### CODE REVIEW

File: HW20/hosts.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+ children:
+  Ubuntu:
+   hosts:
+    192.168.33.13:
+  Centos:
+   hosts:
+    192.168.33.14:
+  Linux:
+   hosts:
+    192.168.33.13:
```
Comment (mavrik90) [inline HW20/hosts.yml]: Do not duplicate host IPs. Use groups instead of hosts.

File: HW20/group_vars/Linux/Linux.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+ansible_user: vagrant
+ansible_ssh_private_key_file: ~/.ssh/id_rsa
+packages:
```
Comment (mavrik90) [inline HW20/group_vars/Linux/Linux.yml:3]: put packages to variable

File: HW20/group_vars/Linux/Linux.yml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+ansible_user: vagrant
+ansible_ssh_private_key_file: ~/.ssh/id_rsa
+packages: 
+  - git
+  - python3
+  - docker-ce
+  - docker-ce-cli
+  - containerd.io
+user: hw20
+shell: /bin/sh
```
Comment (mavrik90) [inline HW20/group_vars/Linux/Linux.yml]: use variable here

Comment (mavrik90) [reply HW20/group_vars/Linux/Linux.yml]: Put all user details into variable and use loop.
For example
```
user_crowd: 
  - username: userlogin
    password: "{{ loginuser_password }}"
    group: "{{ user_gang.name }}"
    uid: 5077
    shell: /bin/bash
    createhome: true
    system: false

File: HW20/roles/hw201/tasks/repo.yml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+- name: Clone git repositories
+  become: true
+  ansible.builtin.git:
+    repo: "{{ repo  }}"
+    dest: /home/"{{  user  }}"
```
Comment (mavrik90) [inline HW20/roles/hw201/tasks/repo.yml]: here should be full path

File: HW20/roles/hw201/tasks/user.yml [hunk 5]
```diff
@@ -0,0 +1,13 @@
+- name: Add user
+  become: true
+  ansible.builtin.user:
```
Comment (mavrik90) [inline HW20/roles/hw201/tasks/user.yml:3]: use loop here

File: HW20/roles/hw201/tasks/user.yml [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+    name: "{{  user  }}"
+    home: "{{  home_dir  }}"
+    shell: "{{  shell  }}"
+    
+- name: Add root
+  become: true
+  ansible.builtin.user:
```
Comment (mavrik90) [inline HW20/roles/hw201/tasks/user.yml:11]: Why not to use add group in previous task?

File: HW20/roles/hw202/tasks/copy.yml [hunk 7]
```diff
@@ -0,0 +1,13 @@
+- name: Copy file
+  become: true
+  ansible.builtin.copy:
```
Comment (mavrik90) [inline HW20/roles/hw202/tasks/copy.yml:3]: use loop here

File: HW20/roles/hw202/tasks/copy.yml [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+    src: "{{  copy_file  }}"
+    dest: "{{  copy_dest  }}"
+
+- name: Add permissions
+  become: true
+  ansible.builtin.file:
```
Comment (mavrik90) [inline HW20/roles/hw202/tasks/copy.yml:10]: Why not to set permissions under https://docs.ansible.com/ansible/latest/collections/ansible/builtin/copy_module.html

File: HW20/roles/hw202/tasks/install.yml [hunk 9]
```diff
@@ -0,0 +1,12 @@
+- name: Install a .deb package
+  become: true
+  apt:
```
Comment (mavrik90) [inline HW20/roles/hw202/tasks/install.yml]: use full path like you have for most modules here. `ansible.builtin.file:` for example

File: HW20/roles/hw202/tasks/template.yml [hunk 10]
```diff
@@ -0,0 +1,5 @@
+- name: Template file
+  become: true
+  ansible.builtin.template:
```
Comment (mavrik90) [inline HW20/roles/hw202/tasks/template.yml:3]: loop here

File: HW20/roles/hw202/tasks/copy.yml [hunk 11]
```diff
@@ -0,0 +1,13 @@
...
+    path: "{{  copy_dest  }}"
+    owner: "{{  user  }}"
+    mode: '0755'
```
Comment (mavrik90) [inline HW20/roles/hw202/tasks/copy.yml]: Set as variable

File: HW20/hosts.yml (hunk 1 above)
Comment (mavrik90) [reply HW20/hosts.yml]: Still needs to be fixed

File: HW20/roles/hw201/tasks/user.yml (hunk 6 above)
Comment (Gritskov-Vitaliy) [reply HW20/roles/hw201/tasks/user.yml:11]: Added as an option. To get root, use the tag

File: HW20/roles/hw202/tasks/copy.yml (hunk 8 above)
Comment (Gritskov-Vitaliy) [reply HW20/roles/hw202/tasks/copy.yml:10]: Added as an option. To get permission, use the tag

File: HW20/group_vars/Linux/Linux.yml (hunk 2 above)
Comment (mavrik90) [reply HW20/group_vars/Linux/Linux.yml:3]: ?

File: HW20/hosts.yml [hunk 12]
```diff
@@ -0,0 +1,10 @@
+all:
```
Comment (mavrik90) [inline HW20/hosts.yml:1]: What is another option can be here to group hosts?

**Review by mavrik90 — APPROVED** [review]
LGTM

File: HW20/hosts.yml (hunk 12 above)
Comment (Gritskov-Vitaliy) [reply HW20/hosts.yml:1]: `ungrouped`

**Pull request merged**